pkg/util/debug_logging.go

```diff
@@ -43,11 +44,15 @@ func MakeDebugLoggerWithReducedRedudancy(writer io.Writer, formatter logrus.Form
 	// setup the logger with debug level logging
 	log := logrus.New()
 	log.Level = logrus.DebugLevel
-	log.Formatter = formatter
+
+	// setup the nilFormatter to allow stifling log entries
+	nilFormatter := newNilFormatter(formatter)
+	log.Formatter = nilFormatter
```

Comment:
My Suggestion here is develop a test case for the log mechanism that specifically resolve the "Data Race" of the default log mechanism.

Reply:
Sounds good, added in 9c0d1fb3d43cf0aa942900daa92f5ead808909a3